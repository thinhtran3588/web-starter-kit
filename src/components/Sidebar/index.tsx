import React from 'react';
import Link from 'next/link';
import { Drawer, List, ListItem, Button } from '@material-ui/core';
import { NavItem } from '@app/core';
import { useStyles } from './styles';

interface Props {
  onClose: () => void;
  open: boolean;
  variant: 'permanent' | 'persistent' | 'temporary' | undefined;
  navItems: NavItem[];
}

export const Sidebar = ({ onClose, open, variant, navItems }: Props): JSX.Element => {
  const classes = useStyles();

  return (
    <Drawer
      anchor='left'
      classes={{
        paper: classes.drawer,
      }}
      onClose={onClose}
      open={open}
      variant={variant}
      data-testid='sidebar'
    >
      <List className={classes.list}>
        {navItems.map((navItem) => (
          <ListItem className={classes.item} disableGutters key={navItem.link}>
            <Link href={navItem.link} key={navItem.link}>
              <Button className={classes.button}>
                <a href={navItem.link} className={classes.link}>
                  {navItem.text}
                </a>
              </Button>
            </Link>
          </ListItem>
        ))}
      </List>
    </Drawer>
  );
};
