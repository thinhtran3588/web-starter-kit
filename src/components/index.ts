export * from './AdminLayout';
export * from './AdminFooter';
export * from './Footer';
export * from './Header';
export * from './Icon';
export * from './Layout';
export * from './Sidebar';
export * from './SidebarMenu';
export * from './FormHeader';
export * from './Field';
export * from './FormFilter';
export * from './FormInput';
export * from './FormSearch';
export * from './Table';
export * from './LanguageSelection';
export * from './Link';
export * from './AuthLayout';
export * from './Head';
export * from './Button';
export * from './Grid';
export * from './Typography';
