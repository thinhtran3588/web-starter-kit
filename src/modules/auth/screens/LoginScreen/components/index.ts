export * from './EmailLogin';
export * from './PhoneNoLogin';
