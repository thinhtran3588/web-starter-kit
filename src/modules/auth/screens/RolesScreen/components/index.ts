export * from './Detail';
export * from './PermissionsTable';
