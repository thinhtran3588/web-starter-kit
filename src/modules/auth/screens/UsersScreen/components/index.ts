export * from './Detail';
export * from './PermissionsTable';
export * from './ChangePassword';
