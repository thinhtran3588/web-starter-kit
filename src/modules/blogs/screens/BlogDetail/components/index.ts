export * from './Content';
export * from './SideMenu';
