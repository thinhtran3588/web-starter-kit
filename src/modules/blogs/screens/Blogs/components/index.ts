export * from './Categories';
export * from './BlogsList';
