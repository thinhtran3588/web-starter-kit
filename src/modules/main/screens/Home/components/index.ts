export * from './CoverImage';
export * from './Features';
export * from './Quote';
export * from './Help';
