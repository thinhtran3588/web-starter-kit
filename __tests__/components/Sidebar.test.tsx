import React from 'react';
import { render } from '@testing-library/react';
import { Sidebar } from '@app/components/Sidebar';

describe('components/Sidebar', () => {
  beforeEach(() => {});

  it('renders successfully', async () => {
    const onClose = jest.fn();
    const { baseElement } = render(<Sidebar onClose={onClose} open={true} variant='temporary' navItems={[]} />);

    expect(baseElement).toMatchSnapshot();
  });
});
