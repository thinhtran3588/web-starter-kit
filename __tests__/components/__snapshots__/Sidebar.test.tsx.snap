// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Sidebar renders successfully 1`] = `
<body
  style="overflow: hidden; padding-right: 0px;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiDrawer-root MuiDrawer-modal"
    data-testid="sidebar"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiPaper-root MuiPaper-elevation16 MuiDrawer-paper makeStyles-drawer-1 MuiDrawer-paperAnchorLeft"
      role="document"
      tabindex="-1"
    >
      <ul
        class="MuiList-root makeStyles-list-2 MuiList-padding"
      />
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
